command/hcl2_upgrade.go

```diff
@@ -562,6 +562,9 @@ func jsonBodyToHCL2Body(out *hclwrite.Body, kvs map[string]interface{}) {
 		case map[string]interface{}:
 			var mostComplexElem interface{}
 			for _, randomElem := range value {
+				if k == "linux_options" || k == "network_interface" {
+					break
+				}
```

Comment:
Because these only contained string values; hcl2_upgrade 'guessed' that this was a map[string]string instead of a block (without an equal sign).

This is quite a bad hack, but the only quick/simple way I can think of. A more complicated way would be to somehow fetch the underlying go-cty/hcl type from the configuration, and then generate HCL from that.

This would generate way more accurate HCL2, but would require a big refactor of `hcl2_upgrade`.